### ISSUE TITLE
Failing to build `abstractclient.cpp` on Arch Linux

### ISSUE THREAD

**Issue by skwerlman** [Cockatrice/Cockatrice#4797]
**System Information:**
OS: Arch Linux
Cockatrice: 1a40102f7168d3c3f2486056e21daf4bcceed331
gcc: 12.2.1-2
Qt: 6.5.0-3.1
Qt packages: `qt6-5compat` `qt6-imageformats` `qt6-svg` `qt6-wayland` `qt6-base`                  `qt6-multimedia` `qt6-tools` `qt6-websockets` `qt6-declarative` `qt6-multimedia-gstreamer` `qt6-translations`

_______________________________________________________________________________________

`abstractclient.cpp` fails to compile:
```
[ 73%] Building CXX object cockatrice/CMakeFiles/cockatrice.dir/src/abstractclient.cpp.o
In file included from /usr/include/qt6/QtCore/qobject.h:18,
                 from /usr/include/qt6/QtCore/QObject:1,
                 from /home/sk/Documents/pkg/cockatrice/cockatrice/src/abstractclient.h:8,
                 from /home/sk/Documents/pkg/cockatrice/cockatrice/src/abstractclient.cpp:1:
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = CommandContainer]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:7:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
 1174 |         static_assert(is_complete<T, void>::value || std::is_void_v<T>,
      |                                             ~~~~~~^~~~~~~~~~~~~~~~~~~~
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<CommandContainer>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = RoomEvent]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:11:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<RoomEvent>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = GameEventContainer]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:12:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<GameEventContainer>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_ServerIdentification]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:13:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_ServerIdentification>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_ConnectionClosed]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:14:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_ConnectionClosed>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_ServerShutdown]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:15:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_ServerShutdown>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_AddToList]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:16:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_AddToList>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_RemoveFromList]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:17:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_RemoveFromList>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_UserJoined]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:18:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_UserJoined>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_UserLeft]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:19:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_UserLeft>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_ServerMessage]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:20:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_ServerMessage>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_ListRooms]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:21:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_ListRooms>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_GameJoined]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:22:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_GameJoined>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_UserMessage]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:23:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_UserMessage>))’ evaluates to false
/usr/include/qt6/QtCore/qmetatype.h: In instantiation of ‘constexpr bool QtPrivate::checkTypeIsSuitableForMetaType() [with X = Event_ReplayAdded]’:
/home/sk/Documents/pkg/cockatrice/cockatrice/src/client_metatypes.h:26:1:   required from here
/usr/include/qt6/QtCore/qmetatype.h:1174:51: error: static assertion failed: Meta Types must be fully defined
/usr/include/qt6/QtCore/qmetatype.h:1174:51: note: ‘(((bool)std::integral_constant<bool, false>::value) || ((bool)std::is_void_v<Event_ReplayAdded>))’ evaluates to false
make[2]: *** [cockatrice/CMakeFiles/cockatrice.dir/build.make:474: cockatrice/CMakeFiles/cockatrice.dir/src/abstractclient.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:323: cockatrice/CMakeFiles/cockatrice.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

CMake output:
```
-- Found CCache /usr/bin/ccache
-- The C compiler identification is GNU 12.2.1
-- The CXX compiler identification is GNU 12.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.40.1") 
-- Commit is not a release or prerelease (no git tag found)
-- Project version: 2.8.1-custom(1a40102)
-- Friendly project version: 2.8.1-custom(1a40102) (2023-04-19)
-- Project version filename: Cockatrice-2.8.1-custom(1a40102)
-- Performing Test CXX_HAS_WARNING_-Wcast-align
-- Performing Test CXX_HAS_WARNING_-Wcast-align - Success
-- Performing Test CXX_HAS_WARNING_-Wmissing-declarations
-- Performing Test CXX_HAS_WARNING_-Wmissing-declarations - Success
-- Performing Test CXX_HAS_WARNING_-Wno-long-long
-- Performing Test CXX_HAS_WARNING_-Wno-long-long - Success
-- Performing Test CXX_HAS_WARNING_-Wno-error=extra
-- Performing Test CXX_HAS_WARNING_-Wno-error=extra - Success
-- Performing Test CXX_HAS_WARNING_-Wno-error=delete-non-virtual-dtor
-- Performing Test CXX_HAS_WARNING_-Wno-error=delete-non-virtual-dtor - Success
-- Performing Test CXX_HAS_WARNING_-Wno-error=sign-compare
-- Performing Test CXX_HAS_WARNING_-Wno-error=sign-compare - Success
-- Performing Test CXX_HAS_WARNING_-Wno-error=missing-declarations
-- Performing Test CXX_HAS_WARNING_-Wno-error=missing-declarations - Success
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Update Translations: OFF
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found OpenGL: /usr/lib/libOpenGL.so   
-- Found Qt 6.5.0 at: /usr/lib/cmake/Qt6
-- Found Protobuf: /usr/lib/libprotobuf.so (found version "3.21.12") 
-- Found Protobuf 3.21.12 at: /usr/lib/libprotobuf.so
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.13")  
-- Looking for lzma_auto_decoder in /usr/lib/liblzma.so
-- Looking for lzma_auto_decoder in /usr/lib/liblzma.so - found
-- Looking for lzma_easy_encoder in /usr/lib/liblzma.so
-- Looking for lzma_easy_encoder in /usr/lib/liblzma.so - found
-- Looking for lzma_lzma_preset in /usr/lib/liblzma.so
-- Looking for lzma_lzma_preset in /usr/lib/liblzma.so - found
-- Found LibLZMA: /usr/lib/liblzma.so (found version "5.4.2") 
-- Configuring done (2.2s)
-- Generating done (0.1s)
-- Build files have been written to: /home/sk/Documents/pkg/cockatrice/build
```

_______________________________________________________________________________________

**Steps to reproduce:**
 - `rm -rf build/`
 - `mkdir build && cd build`
 - `cmake ..`
 - `make -j1`

**Comment by tooomm** [comment]
Can you try again with the newest commit?
Zach just merged some changes from ebbit in regards to Arch.

**Comment by ebbit1q** [comment]
see https://github.com/Cockatrice/Cockatrice/pull/4796

**Comment by skwerlman** [comment]
yep, this is fixed by #4796, thanks
